src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,370 @@
+
+      <_UnManagedResolvedFileToPublish Include="@(ResolvedFileToPublish)"
+                                     Condition="'%(ResolvedFileToPublish.AssetType)'!='runtime'" />
+    </ItemGroup>
+  </Target>
+
+
+  <!--
+    ============================================================
+                                        ComputeFilesToCache
+
+    Gathers all the files that need to be copied to the publish directory.
+    ============================================================
+    -->
+  <Target Name="ComputeFilesToCache"
+          DependsOnTargets="_ComputeNetPublishAssets;
+                            _ComputeCopyToPublishDirectoryItems">
+
+    <PropertyGroup>
+      <CopyBuildOutputToPublishDirectory Condition="'$(CopyBuildOutputToPublishDirectory)'==''">true</CopyBuildOutputToPublishDirectory>
+      <CopyOutputSymbolsToPublishDirectory Condition="'$(CopyOutputSymbolsToPublishDirectory)'==''">true</CopyOutputSymbolsToPublishDirectory>
+    </PropertyGroup>
+
+    <ItemGroup>
+
+      <!-- Copy all the assemblies -->
+      <ResolvedFileToPublish Include="@(ResolvedAssembliesToPublish)">
+      </ResolvedFileToPublish>
+    </ItemGroup>
+
+  </Target>
+
+  <!--
+    ============================================================
+                                        _RunOptimizer
+
+    Start the optimization face
+    ============================================================
+    -->
+  <Target Name="_RunOptimizer"
+          DependsOnTargets="_ComputeResolvedFilesToCachTypes;
+                            _GenerateCrossgenProj;
+                            _SetupCrossgenStage;
+                            _RestoreCrossgen">
+    <!-- Get the coreclr path -->
+    <ItemGroup>
+      <_CoreclrResolvedPath Include="@(CrossgenResolvedAssembliesToPublish)"
+                                             Condition="'%(CrossgenResolvedAssembliesToPublish.Filename)'=='coreclr'" />
+      <_CoreclrResolvedPath Include="@(CrossgenResolvedAssembliesToPublish)"
+                                             Condition="'%(CrossgenResolvedAssembliesToPublish.Filename)'=='libcoreclr'" />
+      <_AssembliestoCrossGen Include="$(_RuntimeDir)\**\*.*" />
+    </ItemGroup>
+
+    <Error Text="Error more than one coreclr found !" Condition="'@(_CoreclrResolvedPath->Count())' &gt; 1" />
+    <Error Text="Error no coreclr found !" Condition="'@(_CoreclrResolvedPath)'== ''" />
+    
+    <!-- Get the crossgen and jit path-->
+    <PropertyGroup>
+      <_CoreclrPath>@(_CoreclrResolvedPath)</_CoreclrPath>
+      <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrPath)))</_CoreclrDir>
+      <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrDir)))</_CoreclrDir>
+      <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrDir)))</_CoreclrDir>
+      <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrDir)))</_CoreclrDir>
```

Comment:
It looks like you're trying to get the relative path `..\..\..\tools` from where the CoreCLR was resolved.  I think you can probably do this much more cleanly with Path.GetFullPath() or a similar method.